binding.c

```diff
@@ -287,38 +287,71 @@ bare_tcp__on_close(uv_handle_t *handle) {
   err = js_get_reference_value(env, tcp->ctx, &ctx);
   assert(err == 0);
 
-  js_value_t *on_close;
-  err = js_get_reference_value(env, tcp->on_close, &on_close);
-  assert(err == 0);
+  if (tcp->resetting) {
+    uv_loop_t *loop;
+    err = js_get_env_loop(env, &loop);
+    assert(err == 0);
 
-  err = js_delete_reference(env, tcp->on_connection);
-  assert(err == 0);
+    uv_tcp_t new_handle;
+    tcp->handle = new_handle;
 
-  err = js_delete_reference(env, tcp->on_connect);
-  assert(err == 0);
+    err = uv_tcp_init(loop, &tcp->handle);
 
-  err = js_delete_reference(env, tcp->on_read);
-  assert(err == 0);
+    if (err < 0) {
+      js_throw_error(env, uv_err_name(err), uv_strerror(err));
```

Comment:
This must be forwarded to the `on_reset` callback as it'll otherwise forcefully trigger an uncaught.